src/serializers/infer.rs

```diff
@@ -198,7 +215,7 @@ pub(crate) fn infer_to_python_known(
                 if let Some(fallback) = extra.fallback {
                     let next_value = fallback.call1((value,))?;
                     let next_result = infer_to_python(next_value, include, exclude, extra);
-                    extra.rec_guard.pop(value_id);
+                    extra.rec_guard.pop(value_id, INFER_DEF_REF_ID);
```

Comment:
would it make sense to change the function signature to `fn pop(&self, id: usize, def_ref_id: Option<usize>)`, in the function use `def_ref_id.unwrap_or(INFER_DEF_REF_ID)` to keep that logic and const in one place?

Reply:
Hmm I like it how it is now.
The way I'm thinking about it is as a two graphs that we're traversing together. So it makes more sense in my head that these integers represent node labels in the respective graphs, with `infer_to_python_known` just being a node that is created at compile time (and hence has a constant label). If we add another similar thing at compile time we can just give it another label and the algorithm will continue to work, if we special case it as `None` then we'd need to change it.

Reply:
But to be clear, I can change it if you feel strongly.

Reply:
no, fine.